src/icalendar/prop.py

```diff
@@ -238,13 +238,22 @@ def from_ical(cls, ical):
 class vDDDLists(object):
     """A list of vDDDTypes values.
     """
-    def __init__(self, dt_list):
+    def __init__(self, dt_list, type_class=None):
+        if type_class is None:
+            type_class = vDDDTypes
         if not hasattr(dt_list, '__iter__'):
             dt_list = [dt_list]
         vDDD = []
         tzid = None
+        if dt_list:
+            # we have some values, all should have the same type
+            ltype = type(dt_list[0])
         for dt in dt_list:
-            dt = vDDDTypes(dt)
+            # raise ValueError if type of the input values differs
+            if not isinstance(dt, ltype):
```

Comment:
Wouldn't this test fail if the first element in the list is a date and the later ones are datetime? I believe we will need to actually do a `if type(dt) != type(ltype)` here.